env0/resource_cost_credentials.go

```diff
@@ -0,0 +1,208 @@
+		ReadContext:   resourceCostCredentialsRead,
+		DeleteContext: resourceCostCredentialsDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "name for the credentials",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"arn": {
+				Type:          schema.TypeString,
+				Description:   "the aws role arn",
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"client_id", "client_secret", "tenant_id", "subscription_id", "table_id", "secret"},
+				ExactlyOneOf:  []string{"client_id", "table_id"},
+			},
+			"external_id": {
+				Type:          schema.TypeString,
+				Description:   "the aws role external id",
+				Optional:      true,
+				Sensitive:     true,
+				ForceNew:      true,
+				RequiredWith:  []string{"arn"},
+				ConflictsWith: []string{"client_id", "client_secret", "tenant_id", "subscription_id", "table_id", "secret"},
+			},
+			"client_id": {
+				Type:          schema.TypeString,
+				Description:   "the azure client id",
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"external_id", "arn", "table_id", "secret"},
+				ExactlyOneOf:  []string{"arn", "table_id"},
+			},
+			"client_secret": {
+				Type:          schema.TypeString,
+				Description:   "azure client secret",
+				Optional:      true,
+				Sensitive:     true,
+				ForceNew:      true,
+				RequiredWith:  []string{"client_id"},
+				ConflictsWith: []string{"external_id", "arn", "table_id", "secret"},
+			},
+			"tenant_id": {
+				Type:          schema.TypeString,
+				Description:   "azure tenant id",
+				Optional:      true,
+				ForceNew:      true,
+				RequiredWith:  []string{"client_id"},
+				ConflictsWith: []string{"external_id", "arn", "table_id", "secret"},
+			},
+			"subscription_id": {
+				Type:          schema.TypeString,
+				Description:   "azure subscription id",
+				Optional:      true,
+				ForceNew:      true,
+				RequiredWith:  []string{"client_id"},
+				ConflictsWith: []string{"external_id", "arn", "table_id", "secret"},
+			},
+			"table_id": {
+				Type:          schema.TypeString,
+				Description:   "the table id of this credentials ",
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"client_id", "client_secret", "tenant_id", "subscription_id", "arn", "external_id"},
+				ExactlyOneOf:  []string{"arn", "client_id"},
+			},
+			"secret": {
+				Type:          schema.TypeString,
+				Description:   "the secret of this credentials",
+				Optional:      true,
+				Sensitive:     true,
+				ForceNew:      true,
+				RequiredWith:  []string{"table_id"},
+				ConflictsWith: []string{"client_id", "client_secret", "tenant_id", "subscription_id", "arn", "external_id"},
+			},
+		},
+	}
+}
+
+func resourceCostCredentialsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	apiClient := meta.(ApiClientInterface)
+	var apikey client.Credentials
```

Comment:
in `apikey` the k should be in upper case because it's another word, like you did in `apiClient`